evm-template/runtime/src/configs/asset_config.rs

```diff
@@ -184,3 +184,41 @@ impl pallet_asset_manager::Config for Runtime {
     /// Rerun benchmarks if you are making changes to runtime configuration.
     type WeightInfo = weights::pallet_asset_manager::WeightInfo<Runtime>;
 }
+
+/// This trait ensure we can convert AccountIds to AssetIds.
+pub trait AccountIdAssetIdConversion<Account, AssetId> {
+    // Get assetId and prefix from account
+    fn account_to_asset_id(account: Account) -> Option<(Vec<u8>, AssetId)>;
+
+    // Get AccountId from AssetId and prefix
+    fn asset_id_to_account(prefix: &[u8], asset_id: AssetId) -> Account;
+}
+
+const FOREIGN_ASSET_PRECOMPILE_ADDRESS_PREFIX: &[u8] = &[255u8; 4];
```

Comment:
it is complicated, the whole trait can be eliminated imo, but I'll explain that later possibly in a sync call. Right now let's keep it because it creates other weird problems (removal of the trait)